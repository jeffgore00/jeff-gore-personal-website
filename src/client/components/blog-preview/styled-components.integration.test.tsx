/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import {
  BlogPreviewTitleHeading,
  BlogPreviewWrapper,
  BlogPreviewTypeHeading,
} from './styled-components';

describe('BlogPreviewTitleHeading presentational component', () => {
  it('renders the wrapped content with the expected CSS styles and HTML attributes', () => {
    const { container } = render(
      <BlogPreviewTitleHeading>
        The Algorithms That Still Matter
      </BlogPreviewTitleHeading>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('BlogPreviewWrapper presentational component', () => {
  it('renders the wrapped content with the expected CSS styles and HTML attributes', () => {
    const { container } = render(
      <BlogPreviewWrapper $contentId="20500402-DUMMY-the-algorithms-that-still-matter">
        <div className="blog-preview-title-heading">
          The Algorithms That Still Matter
        </div>
        <span className="blog-preview-subtitle">
          A cheat sheet to some fundamentals that are older than me.
        </span>
      </BlogPreviewWrapper>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('BlogPreviewTypeHeading presentational component', () => {
  describe('When the blog type is "TECH"', () => {
    it('renders the blogType with the expected CSS styles (incl. blue font) and HTML attributes', () => {
      const { container } = render(<BlogPreviewTypeHeading blogType="TECH" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('When the blog type is not "TECH"', () => {
    it('renders the blogType with the expected CSS styles (incl. purple font) and HTML attributes', () => {
      const { container } = render(
        <BlogPreviewTypeHeading blogType="COMMENTARY" />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
