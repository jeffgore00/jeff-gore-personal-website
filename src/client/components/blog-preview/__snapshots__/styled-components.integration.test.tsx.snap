// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPreviewTitleHeading presentational component renders the wrapped content with the expected CSS styles and HTML attributes 1`] = `
.c0 {
  font-weight: bold;
  margin-bottom: 0;
}

<div
  class="c0 blog-preview-title-heading"
>
  The Algorithms That Still Matter
</div>
`;

exports[`BlogPreviewTypeHeading presentational component When the blog type is "TECH" renders the blogType with the expected CSS styles (incl. blue font) and HTML attributes 1`] = `
.c0 {
  margin-bottom: 0;
}

<div
  class="c0 blog-preview-type-heading"
>
  <span
    style="font-weight: bold; color: blue;"
  >
    TECH
  </span>
</div>
`;

exports[`BlogPreviewTypeHeading presentational component When the blog type is not "TECH" renders the blogType with the expected CSS styles (incl. purple font) and HTML attributes 1`] = `
.c0 {
  margin-bottom: 0;
}

<div
  class="c0 blog-preview-type-heading"
>
  <span
    style="font-weight: bold; color: purple;"
  >
    COMMENTARY
  </span>
</div>
`;

exports[`BlogPreviewWrapper presentational component renders the wrapped content with the expected CSS styles and HTML attributes 1`] = `
.c0 {
  margin-bottom: 1em;
}

<div
  class="c0 single-blog-preview"
  id="blog-preview-20500402-DUMMY-the-algorithms-that-still-matter"
>
  <div
    class="blog-preview-title-heading"
  >
    The Algorithms That Still Matter
  </div>
  <span
    class="blog-preview-subtitle"
  >
    A cheat sheet to some fundamentals that are older than me.
  </span>
</div>
`;
